packages/node-fetch/src/URL.ts

```diff
@@ -19,13 +19,22 @@ export class PonyfillURL extends FastUrl implements URL {
       return;
     }
     this.parse(url, false);
-    // `fast-url-parser` cannot handle ipv6 hosts correctly
+    // `fast-url-parser` incorrectly removes `[IPV6]` in the hostname
+    // Then `hostname` doesn't include `[` and `]` for IPv6 URLs
```

Comment:
are there open issues on the original implementations and environments that cover that things we need to fix in this library?

Reply:
yes but as we ponyfill, we have found this issue right?
where is the source of the issue?

Reply:
The issue is in `fast-url-parser` originally.

Reply:
maybe I need also more general context about the `fast-url-parser` library
seems like a very old library, also we are using a fork by Kamil
in general, what's the story there and aren't there more alternatives for it?

Reply:
Kamil's fork fixes this https://github.com/petkaantonov/urlparser/issues/20
We started using it to improve the performance of url parsing.
However, Node introduced new URL parser based on Ada, but we haven't had a chance to test if it is as fast as we need.

Reply:
let's see if we can find a better alternative that is more used and follows all the test cases here and is faster